source/deprecation-policy.rst

```diff
@@ -0,0 +1,145 @@
+the ``@deprecated`` annotation. To have usages of functionality in your dependencies analyzed
+and reported, you must use a tool like `PHPStan Deprecation Rules`_ or `Pslam Static Analysis`_.
+
+Here is an example of how this would be used in Doctrine:
+
+.. code-block:: php
+
+    namespace Doctrine\ORM;
+
+    class EntityManager
+    {
+        // ...
+
+        /**
+         * @deprecated
+         */
+        public function getSomething() : int
+        {
+            // ...
+        }
+    }
+
+Then, if your code directly depends on ``doctrine/orm`` which provides the
+``Doctrine\ORM\EntityManager`` class, you will get a warning about it when you run
+`PHPStan Deprecation Rules`_ or `Pslam Static Analysis`_. Here is an example:
+
+.. code-block:: php
+
+    namespace App;
+
+    use App\Entities\User;
+    use Doctrine\ORM\EntityManager;
+
+    class UserRegistration
+    {
+        /** @var EntityManager */
+        private $entityManager;
+
+        // ...
+
+        public function create(array $userData) : User
+        {
+            $something = $this->entityManager->getSomething();
+
+            // ...
+        }
+    }
+
+Using ``trigger_error()`` and ``phpunit-bridge``
+------------------------------------------------
+
+In some cases, you may need to conditionally deprecate functionality. In these cases, it is required
+to use the ``trigger_error()`` function and a tool like `PHPUnit Bridge`_ which implements a custom
+error handler for you to use to collect and report these warnings. This method is not recommended
+and Doctrine will try to avoid using it if possible.
+
+If you do not wish to use `PHPUnit Bridge`_ you can implement your own custom error handler using
+the `set_error_handler`_ function.
+
+.. note::
+
+    In order for this strategy to work, you have to configure your `error_reporting`_ properly
+    in both your development and production environments. You don't want errors of type ``E_USER_DEPRECATED``
+    being reported from PHP.
+
+Here is an example of how this would be used in Doctrine:
+
+.. code-block:: php
+
+    namespace Doctrine\ORM;
+
+    class EntityManager
+    {
+        // ...
+
+        public function getAnotherThing() : int
+        {
+            if ($this->featureFlag) {
+                trigger_error('This feature has been disabled.', E_USER_DEPRECATED);
+            }
+
+            // ...
+        }
+    }
+
+Now in order to get those warnings reported to you in your automation, you need to use a tool like `PHPUnit Bridge`_.
+
+Tools
+-----
+
+There are a few tools out there that you can use to integrate deprecation warnings in to your
+automation.
+
+PHPStan Deprecation Rules
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Doctrine uses and recommends PHPStan_ for reporting usages of deprecated functionality in your code.
+
+.. code-block:: console
+
+    $ composer require --dev phpstan/phpstan
+    $ composer require --dev phpstan/phpstan-deprecation-rules
```

Comment:
Should we provide a sample config file? The extension doesn't get automatically enabled by just installing it via composer, one needs to append the extension file to the configuration, like https://github.com/lcobucci/kafka/blob/1b1284020f20018e2d54db2d97ee30e91dff7708/phpstan.neon.dist#L5